vortex-serde/src/layouts/read/layouts/column.rs

```diff
@@ -47,98 +52,523 @@ impl LayoutSpec for ColumnLayoutSpec {
+            vortex_bail!("Column layout must have struct dtype")
+        };
+
+        let mut unhandled_children_names = Vec::new();
+        let mut unhandled_children = Vec::new();
+        let mut handled_children = Vec::new();
+        let mut handled_names = Vec::new();
+
+        for (field, (name, dtype)) in refs
+            .into_iter()
+            .zip_eq(s.names().iter().cloned().zip_eq(s.dtypes().iter().cloned()))
+        {
+            let resolved_child = lazy_dtype.resolve_field(&field)?;
+            let child_loc = fb_children.get(resolved_child)._tab.loc();
+            let filter = self
+                .scan
+                .expr
+                .as_ref()
+                .and_then(|e| filter_project(e, &[field]));
+
+            let handled =
+                self.scan.expr.is_none() || (self.scan.expr.is_some() && filter.is_some());
 
-    pub fn lazy_dtype(&self) -> VortexResult<Arc<LazyDeserializedDType>> {
-        match &self.scan.projection {
-            Projection::All => Ok(self.message_cache.dtype().clone()),
-            Projection::Flat(p) => self.message_cache.dtype().project(p),
+            let mut child = self.layout_serde.read_layout(
+                self.fb_bytes.clone(),
+                child_loc,
+                self.length,
+                Scan::new(filter),
+                self.message_cache.relative(
+                    resolved_child as u16,
+                    Arc::new(LazyDeserializedDType::from_dtype(dtype)),
+                ),
+            )?;
+
+            if self.offset != 0 {
+                child.advance(self.offset)?;
+            }
+
+            if handled {
+                handled_children.push(child);
+                handled_names.push(name);
+            } else {
+                unhandled_children.push(child);
+                unhandled_children_names.push(name);
+            }
+        }
+
+        if !unhandled_children_names.is_empty() {
+            let prf = self
+                .scan
+                .expr
+                .as_ref()
+                .and_then(|e| {
+                    filter_project(
+                        e,
+                        &unhandled_children_names
+                            .iter()
+                            .map(|n| Field::from(n.as_ref()))
+                            .collect::<Vec<_>>(),
+                    )
+                })
+                .ok_or_else(|| {
+                    vortex_err!(
+                        "Must be able to project {:?} filter into unhandled space {}",
+                        self.scan.expr.as_ref(),
+                        unhandled_children_names.iter().format(",")
+                    )
+                })?;
+
+            handled_children.push(Box::new(ColumnBatchReader::new(
+                unhandled_children_names.into(),
+                unhandled_children,
+                Some(prf),
+                false,
+            )));
+            handled_names.push("unhandled".into());
```

Comment:
This name actually is not reserved. In cases where it would conflict it will not hit this codepath and in all other cases we do not look at the columns names when evaluating expression so while it could conflict we don't resolve columns by their names. However, it seems prudent to make it be something explicit